docs/ballerina-to-oas/spec/spec.md

```diff
@@ -725,6 +725,26 @@ parameters:
       nullable: true
 ```
 
+The query parameter name in the schema is defaulted to the Ballerina parameter name. But this can be overridden using the `@http:Query` annotation.
+
+```ballerina
+service /api on new http:Listener(9090) {
+
+  resource function get path(@http:Query{name: "userName"} string param) {
+      // ...
+  }
+}
+```
+
+```yml
+parameters:
+  - name: userName
```

Comment:
It needs to be mentioned that `param` name details with `x-ballerina-name` in here, don't we?

Reply:
No, the extension is not related to ballerina to openapi specification.

Reply:
Yeah, that makes sense, I see two options:

**Option 1**: We can include all modified and overridden names in the OAS. This way, when generating a service or client, we won’t need to worry about creating Ballerina-friendly names because we can simply pull them from the OAS.

**Option 2**: We keep the OAS clean by not adding any Ballerina-specific details, making it more universal and suitable for any programming language. Instead, we can append these extension details to our current implementation, which already handles mapping package specifics and includes Ballerina-specific details.

I prefer Option 2 to handle ballerina extension details with package detail implementations, WDYT?

Reply:
Yes, I also prefer Option 2 to keep the OAS clean rather than adding Ballerina specific extensions. Users can always sanitize the OAS, if they want to generate Ballerina client or service.

Reply:
Shall we create a separate issue to track the extensions details adding? and let's keep our default behaviour as we fixed now.